feat: add distinctField for RS API (#184)
appbaseio/Docs: content/docs/search/reactivesearch-api/reference.md

}
```

### distinctField

> You can use the `distinctField` parameter to collapse search results based on field values. The collapsing is done by selecting only the top sorted document per collapse key. You can read more about it [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/collapse-search-results.html).

Reviewer: ```suggestion
This property returns only the distinct value documents for the specified field. It is equivalent to the `DISTINCT` clause in SQL. It internally uses the collapse feature of Elasticsearch. You can read more about it over [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/collapse-search-results.html).
```

| Type     | Applicable on query of type | Required |
| ------   | --------------------------- | -------- |
| `string` | `all`                  | false    |

The following query would return the products for distinct brands.
```js
{
    "query": [
        {
            "id": "test",
            "dataField": [
                "product_name"
            ],
            "distinctField": "brand.keyword",
        }
    ]
}
```

### distinctFieldConfig

> This property allows expanding each collapsed top hits with the inner_hits option.

Reviewer: ```suggestion
This property allows specifying additional options to the `distinctField` property. Using the allowed DSL, one can specify how to return K distinct values (default value of K=1), sort them by a specific order, or return a second level of distinct values. `distinctFieldConfig` object corresponds to the `inner_hits` key's DSL. You can read more about it over [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/collapse-search-results.html).
```

| Type     | Applicable on query of type | Required |
| ------   | --------------------------- | -------- |

> | `string` | `all`                       | false    |

Reviewer: @bietkul This is an object type, not string.

> | `string` | `all`                       | false    |

Author: ```suggestion
| `object` | `all`                       | false    |
```
